### PR TITLE
Snmpreceiver fix counter64 

### PR DESCRIPTION
**Pull request by technimad-splunk**
**Description:**
SNMP data with type Counter64 can be sent as a uint64. This type wasn't handled.
Internally we convert all numerical SNMP values to an int64.
The conversion for a uint64, sent by the newly supported Conter64 SNMP data type, to int64
was lacking and is introduced with this PR.

**Link to tracking Issue:** 
#23897, PR #26119

**Testing:** 
Ran the collector with a config to collect oid 1.3.6.1.2.1.31.1.1.1.6, which is a Counter64.
This didn't work, showed value type as unsupported.
After the change data from this oid comes in as expected.

**Documentation:** n/a

### CODE REVIEW

File: receiver/snmpreceiver/client.go [hunk 1]
```diff
@@ -283,6 +283,7 @@ func (c *snmpClient) GetIndexedData(oids []string, scraperErrors *scrapererror.S
 			}
 			// Convert data into the more simplified data type
 			clientSNMPData := c.convertSnmpPDUToSnmpData(snmpPDU)
+
```
Comment (atoulme) [inline receiver/snmpreceiver/client.go:286]:
```suggestion
```

**Comment by fatsheep9146** [comment]
failed unit test should be fixed @technimad-splunk

**Comment by technimad-splunk** [comment]
I'm still working on the tests.

**Comment by technimad-splunk** [comment]
> failed unit test should be fixed @technimad-splunk

Test are green now. Fixed by ed97de1

**Review by fatsheep9146 — APPROVED** [review]

File: receiver/snmpreceiver/client_test.go [hunk 2]
```diff
@@ -513,7 +513,7 @@ func TestGetScalarData(t *testing.T) {
 				expectedSNMPData := []SNMPData{}
 				mockGoSNMP := new(mocks.MockGoSNMPWrapper)
 				pdu1 := gosnmp.SnmpPDU{
-					Value: uint64(math.MaxUint64),
+					Value: float64(math.MaxFloat64),
```
[the remark this answers is not in the document]
Comment (djaglowski) [reply receiver/snmpreceiver/client_test.go:516]: Ah, missed the context. Thanks for explaining.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**